HOMEWORK_30/FinalProject/Platform.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+#include <SFML/Graphics.hpp>
+#include <vector>
+#include <memory>
+
+class Platform 
+{
+public:
+    sf::Sprite sprite;
+    float x, y;
+
+    Platform();
+    virtual ~Platform();
+
+    virtual bool isBrokenPlatform() const = 0;
+    virtual bool isFalling() const = 0;
+    virtual void startFalling() = 0;
```

Comment:
This is not what inheritance should be used for(

isBrokenPlatform part should be only a part of BrokenPlatform, not general interface. Please revisit `SOLID` principles once again and try to see how your Platform design violates that

Reply:
In short, these details should be left in subclasses, e.g. in update method. So outside you just call `for( platform  :platforms) platform->update(dt);` and that's it. Each specific Platform class does this update differently. Some gets destroyed, other is moving, etc.